frb_rust/src/handler.rs

```diff
@@ -204,31 +202,44 @@ impl<EH: ErrorHandler> ThreadPoolExecutor<EH> {
 }
 
 impl<EH: ErrorHandler> Executor for ThreadPoolExecutor<EH> {
+    #[allow(clippy::clone_on_copy, clippy::redundant_clone)]
     fn execute<TaskFn, TaskRet>(&self, wrap_info: WrapInfo, task: TaskFn)
     where
         TaskFn: FnOnce(TaskCallback) -> Result<TaskRet> + Send + UnwindSafe + 'static,
         TaskRet: IntoDart,
     {
         const NUM_WORKERS: usize = 4;
-        lazy_static! {
-            static ref THREAD_POOL: Mutex<ThreadPool> = Mutex::new(ThreadPool::with_name(
-                "frb_executor".to_string(),
-                NUM_WORKERS
-            ));
+        #[cfg(not(target_family = "wasm"))]
+        lazy_static::lazy_static! {
+            static ref THREAD_POOL: parking_lot::Mutex<threadpool::ThreadPool> =
+                parking_lot::Mutex::new(threadpool::ThreadPool::with_name(
+                    "frb_executor".to_string(),
+                    NUM_WORKERS
+                ));
+        }
+
+        #[cfg(target_family = "wasm")]
+        thread_local! {
+            static WORKER_POOL: crate::pool::WorkerPool = crate::pool::WorkerPool::new(
+                NUM_WORKERS, std::option_env!("FRB_JS").expect("FRB_JS not set").into()
```

Comment:
TODO: Find a better way to do this

Reply:
If you mean that env var and it is only used for debug names then I think it is ok (since only for debugging purpose)

Reply:
This is getting the location of the output JS/WASM files *before* actually building them, so this needs a bit of reorganizing to solve.